Tests/RegexTests/ParseTests.swift

```diff
@@ -210,9 +210,13 @@ extension RegexTests {
     parseTest(#"\08"#, concat(scalar("\u{0}"), "8"))
     parseTest(#"\0707"#, concat(scalar("\u{38}"), "7"))
 
-    parseTest(#"[\0]"#, charClass("\u{0}"))
-    parseTest(#"[\01]"#, charClass("\u{1}"))
-    parseTest(#"[\070]"#, charClass("\u{38}"))
+    // FIXME(Hamish): These now get printed using the unicode
+    // literal syntax instead of rendered as Character. Adjust
+    // testing infra to handle that.
+//    parseTest(#"[\0]"#, charClass("\u{0}"))
+//    parseTest(#"[\01]"#, charClass("\u{1}"))
+//    parseTest(#"[\070]"#, charClass("\u{38}"))
+
```

Comment:
@hamishknight I changed the AST dump functions to print this out as `\u{0}` instead of inserting a literal null byte, so some tests fail. AFAICT, this is just a testing infra concern and not a observable behavioral change.

Reply:
Ah I think these will need to be e.g `charClass(scalar_m("\u{0}"))` instead of `charClass("\u{0}")`